modules/router-store/src/router_store_module.ts

```diff
@@ -310,29 +313,31 @@ export class StoreRouterConnectingModule {
   }
 
   private setUpStoreStateListener(): void {
-    this.store.subscribe(state => {
-      this.storeState = state;
-    });
-    this.store.pipe(select(this.stateKey)).subscribe(() => {
-      this.navigateIfNeeded();
-    });
+    this.store
+      .pipe(select(this.stateKey), withLatestFrom(this.store))
+      .subscribe(([routerStoreState, storeState]) => {
+        this.navigateIfNeeded(routerStoreState, storeState);
+      });
   }
 
-  private navigateIfNeeded(): void {
-    if (
-      !this.storeState[this.stateKey] ||
-      !this.storeState[this.stateKey].state
-    ) {
+  private navigateIfNeeded(
+    routerStoreState: RouterReducerState,
+    storeState: any
+  ): void {
+    if (!routerStoreState || !routerStoreState.state) {
       return;
     }
     if (this.trigger === RouterTrigger.ROUTER) {
       return;
     }
 
-    const url = this.storeState[this.stateKey].state.url;
+    const url = routerStoreState.state.url;
     if (this.router.url !== url) {
+      this.storeState = storeState;
       this.trigger = RouterTrigger.STORE;
-      this.router.navigateByUrl(url);
+      this.router.navigateByUrl(url).catch(e => {
+        this.errorHandler.handleError(e);
```

Comment:
we should write a test for this.

Reply:
There already is, kind of, the "...if navigation errors (navigation initiated by store)"-test. I will enhance it to check the error handler is invoked correctly.